core/src/components/button/button.ionic.scss

```
@@ -246,26 +267,32 @@
   font-size: globals.$ion-font-size-500;
 }
 
-:host(.button-small) ::slotted(ion-icon[slot="start"]) {
+:host(.button-small) ::slotted(ion-icon[slot="start"]),
+::slotted(ion-spinner[slot="start"]) {
```

```suggestion
:host(.button-small) ::slotted(ion-spinner[slot="start"]) {
```

These aren't being set correctly and are being overwritten by the last one, in this case large. Please make sure to verify the styles are being applied correctly before ready for review.